pkg/asset/installconfig/gcp/gcp.go

```diff
@@ -82,13 +82,40 @@ func selectProject(ctx context.Context) (string, error) {
 	return selectedProject, err
 }
 
-func selectRegion(project string) (string, error) {
-	longRegions := make([]string, 0, len(validation.Regions))
-	shortRegions := make([]string, 0, len(validation.Regions))
-	for id, location := range validation.Regions {
-		longRegions = append(longRegions, fmt.Sprintf("%s (%s)", id, location))
-		shortRegions = append(shortRegions, id)
+func selectRegion(ctx context.Context, project string) (string, error) {
+
+	ssn, err := GetSession(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get session")
+	}
+
+	client := &Client{
+		ssn: ssn,
+	}
+
+	validRegions, err := client.GetProjectRegions(ctx, project)
+	if err != nil {
+		return "", err
 	}
+
+	longRegions := make([]string, 0, len(validRegions))
+	shortRegions := make([]string, 0, len(validRegions))
+	for i := range validRegions {
+		found := false
+		for id, location := range validation.Regions {
+			if id == validRegions[i] {
+				longRegions = append(longRegions, fmt.Sprintf("%s (%s)", id, location))
+				shortRegions = append(shortRegions, id)
+				found = true
+				break
+			}
+		}
+		if !found {
+			longRegions = append(longRegions, fmt.Sprintf("%s", validRegions[i]))
+			shortRegions = append(shortRegions, validRegions[i])
+		}
+	}
```

Comment:
Would it make more sense for `GetProjectRegions` to return both the short and long names? perhaps as a map[shortregions]longRegions?

Reply:
That was a thought, but I was allowing the caller to alter the data as they pleased. Technically the long name may not exist. Is that something that we still want returned in a map?